eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -485,11 +465,28 @@ public function buildLocation(
         );
     }
 
+    /**
+     * @param \eZ\Publish\SPI\Persistence\Content\Location $spiLocation
+     * @param \eZ\Publish\API\Repository\Values\Content\Content|null $content
+     * @param \eZ\Publish\SPI\Persistence\Content\ContentInfo|null $spiContentInfo
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Location
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     */
     public function buildLocationWithContent(
         SPILocation $spiLocation,
-        APIContent $content,
-        SPIContentInfo $spiContentInfo = null
+        ?APIContent $content,
+        ?SPIContentInfo $spiContentInfo = null
```

Comment:
i don't understand this. why we default `$spiContentinfo` to null but we don't default `$content` to null also?

Reply:
Hi @crevillo! I decided to do so because there is exactly one case when the `$content` parameter can be null: root location which doesn't point to any content. Ref. https://github.com/ezsystems/ezpublish-kernel/pull/2560/files#diff-4043c8073a9a8c2593b809f674fafa42R487